lmdeploy/pytorch/config.py

```diff
@@ -89,6 +89,35 @@ def __post_init__(self):
             self.enable_prefix_caching = False
 
 
+@dataclass
+class DistConfig:
+    dp: int = 1
+    tp: int = 1
+    ep: int = 1
+    dp_rank: int = 0
+    world_size: int = None
+    attn_config: 'DistConfig' = None
+
+    def __post_init__(self):
+        """post init."""
+        assert self.dp_rank < self.dp
+        assert self.dp >= 1
+        if self.dp == 1:
+            world_size = max(self.tp, self.ep)
+            attn_config = self
+        else:
+            world_size = self.dp
+            attn_config = DistConfig(dp=1, tp=1, ep=1, dp_rank=0)
+        self.world_size = world_size
+        self.attn_config = attn_config
+
+    def need_dummy_batch(self):
+        """need dummy batch."""
+        if self.dp == 1:
+            return False
+        return self.tp > 1 #or self.ep > 1
```

Comment:
dummy batch is created if there is no data in one of the dp rank. I think ep>1 needs dummy batch.